// material-dashboard-react/dist/assets/jss/material-dashboard-react/components/headerStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _materialDashboardReact = require("assets/jss/material-dashboard-react.jsx");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var headerStyle = function headerStyle() {
  return {
    appBar: {
      backgroundColor: "transparent",
      boxShadow: "none",
      borderBottom: "0",
      marginBottom: "0",
      position: "absolute",
      width: "100%",
      paddingTop: "10px",
      zIndex: "1029",
      color: _materialDashboardReact.grayColor[7],
      border: "0",
      borderRadius: "3px",
      padding: "10px 0",
      transition: "all 150ms ease 0s",
      minHeight: "50px",
      display: "block"
    },
    container: _objectSpread({}, _materialDashboardReact.container, {
      minHeight: "50px"
    }),
    flex: {
      flex: 1
    },
    title: _objectSpread({}, _materialDashboardReact.defaultFont, {
      letterSpacing: "unset",
      lineHeight: "30px",
      fontSize: "18px",
      borderRadius: "3px",
      textTransform: "none",
      color: "inherit",
      margin: "0",
      "&:hover,&:focus": {
        background: "transparent"
      }
    }),
    appResponsive: {
      top: "8px"
    },
    primary: _objectSpread({
      backgroundColor: _materialDashboardReact.primaryColor[0],
      color: _materialDashboardReact.whiteColor
    }, _materialDashboardReact.defaultBoxShadow),
    info: _objectSpread({
      backgroundColor: _materialDashboardReact.infoColor[0],
      color: _materialDashboardReact.whiteColor
    }, _materialDashboardReact.defaultBoxShadow),
    success: _objectSpread({
      backgroundColor: _materialDashboardReact.successColor[0],
      color: _materialDashboardReact.whiteColor
    }, _materialDashboardReact.defaultBoxShadow),
    warning: _objectSpread({
      backgroundColor: _materialDashboardReact.warningColor[0],
      color: _materialDashboardReact.whiteColor
    }, _materialDashboardReact.defaultBoxShadow),
    danger: _objectSpread({
      backgroundColor: _materialDashboardReact.dangerColor[0],
      color: _materialDashboardReact.whiteColor
    }, _materialDashboardReact.defaultBoxShadow)
  };
};

var _default = headerStyle;
exports.default = _default;
