// material-dashboard-react/dist/assets/jss/material-dashboard-react/components/snackbarContentStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _materialDashboardReact = require("assets/jss/material-dashboard-react.jsx");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var snackbarContentStyle = {
  root: _objectSpread({}, _materialDashboardReact.defaultFont, {
    flexWrap: "unset",
    position: "relative",
    padding: "20px 15px",
    lineHeight: "20px",
    marginBottom: "20px",
    fontSize: "14px",
    backgroundColor: _materialDashboardReact.whiteColor,
    color: _materialDashboardReact.grayColor[7],
    borderRadius: "3px",
    minWidth: "unset",
    maxWidth: "unset",
    boxShadow: "0 12px 20px -10px rgba(" + (0, _materialDashboardReact.hexToRgb)(_materialDashboardReact.whiteColor) + ", 0.28), 0 4px 20px 0px rgba(" + (0, _materialDashboardReact.hexToRgb)(_materialDashboardReact.blackColor) + ", 0.12), 0 7px 8px -5px rgba(" + (0, _materialDashboardReact.hexToRgb)(_materialDashboardReact.whiteColor) + ", 0.2)"
  }),
  top20: {
    top: "20px"
  },
  top40: {
    top: "40px"
  },
  info: _objectSpread({
    backgroundColor: _materialDashboardReact.infoColor[3],
    color: _materialDashboardReact.whiteColor
  }, _materialDashboardReact.infoBoxShadow),
  success: _objectSpread({
    backgroundColor: _materialDashboardReact.successColor[3],
    color: _materialDashboardReact.whiteColor
  }, _materialDashboardReact.successBoxShadow),
  warning: _objectSpread({
    backgroundColor: _materialDashboardReact.warningColor[3],
    color: _materialDashboardReact.whiteColor
  }, _materialDashboardReact.warningBoxShadow),
  danger: _objectSpread({
    backgroundColor: _materialDashboardReact.dangerColor[3],
    color: _materialDashboardReact.whiteColor
  }, _materialDashboardReact.dangerBoxShadow),
  primary: _objectSpread({
    backgroundColor: _materialDashboardReact.primaryColor[3],
    color: _materialDashboardReact.whiteColor
  }, _materialDashboardReact.primaryBoxShadow),
  rose: _objectSpread({
    backgroundColor: _materialDashboardReact.roseColor[3],
    color: _materialDashboardReact.whiteColor
  }, _materialDashboardReact.roseBoxShadow),
  message: {
    padding: "0",
    display: "block",
    maxWidth: "89%"
  },
  close: {
    width: "11px",
    height: "11px"
  },
  iconButton: {
    width: "24px",
    height: "24px",
    padding: "0px"
  },
  icon: {
    display: "block",
    left: "15px",
    position: "absolute",
    top: "50%",
    marginTop: "-15px",
    width: "30px",
    height: "30px"
  },
  infoIcon: {
    color: _materialDashboardReact.infoColor[3]
  },
  successIcon: {
    color: _materialDashboardReact.successColor[3]
  },
  warningIcon: {
    color: _materialDashboardReact.warningColor[3]
  },
  dangerIcon: {
    color: _materialDashboardReact.dangerColor[3]
  },
  primaryIcon: {
    color: _materialDashboardReact.primaryColor[3]
  },
  roseIcon: {
    color: _materialDashboardReact.roseColor[3]
  },
  iconMessage: {
    paddingLeft: "50px",
    display: "block"
  },
  actionRTL: {
    marginLeft: "-8px",
    marginRight: "auto"
  }
};
var _default = snackbarContentStyle;
exports.default = _default;
