// material-dashboard-react/dist/assets/jss/material-dashboard-react/components/tableStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _materialDashboardReact = require("assets/jss/material-dashboard-react.jsx");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var tableStyle = function tableStyle(theme) {
  return {
    warningTableHeader: {
      color: _materialDashboardReact.warningColor[0]
    },
    primaryTableHeader: {
      color: _materialDashboardReact.primaryColor[0]
    },
    dangerTableHeader: {
      color: _materialDashboardReact.dangerColor[0]
    },
    successTableHeader: {
      color: _materialDashboardReact.successColor[0]
    },
    infoTableHeader: {
      color: _materialDashboardReact.infoColor[0]
    },
    roseTableHeader: {
      color: _materialDashboardReact.roseColor[0]
    },
    grayTableHeader: {
      color: _materialDashboardReact.grayColor[0]
    },
    table: {
      marginBottom: "0",
      width: "100%",
      maxWidth: "100%",
      backgroundColor: "transparent",
      borderSpacing: "0",
      borderCollapse: "collapse"
    },
    tableHeadCell: _objectSpread({
      color: "inherit"
    }, _materialDashboardReact.defaultFont, {
      "&, &$tableCell": {
        fontSize: "1em"
      }
    }),
    tableCell: _objectSpread({}, _materialDashboardReact.defaultFont, {
      lineHeight: "1.42857143",
      padding: "12px 8px",
      verticalAlign: "middle",
      fontSize: "0.8125rem"
    }),
    tableResponsive: {
      width: "100%",
      marginTop: theme.spacing(3),
      overflowX: "auto"
    },
    tableHeadRow: {
      height: "56px",
      color: "inherit",
      display: "table-row",
      outline: "none",
      verticalAlign: "middle"
    },
    tableBodyRow: {
      height: "48px",
      color: "inherit",
      display: "table-row",
      outline: "none",
      verticalAlign: "middle"
    }
  };
};

var _default = tableStyle;
exports.default = _default;
