// material-dashboard-react/dist/assets/jss/material-dashboard-react/components/typographyStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _materialDashboardReact = require("assets/jss/material-dashboard-react.jsx");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var typographyStyle = {
  defaultFontStyle: _objectSpread({}, _materialDashboardReact.defaultFont, {
    fontSize: "14px"
  }),
  defaultHeaderMargins: {
    marginTop: "20px",
    marginBottom: "10px"
  },
  quote: {
    padding: "10px 20px",
    margin: "0 0 20px",
    fontSize: "17.5px",
    borderLeft: "5px solid " + _materialDashboardReact.grayColor[10]
  },
  quoteText: {
    margin: "0 0 10px",
    fontStyle: "italic"
  },
  quoteAuthor: {
    display: "block",
    fontSize: "80%",
    lineHeight: "1.42857143",
    color: _materialDashboardReact.grayColor[1]
  },
  mutedText: {
    color: _materialDashboardReact.grayColor[1]
  },
  primaryText: {
    color: _materialDashboardReact.primaryColor[0]
  },
  infoText: {
    color: _materialDashboardReact.infoColor[0]
  },
  successText: {
    color: _materialDashboardReact.successColor[0]
  },
  warningText: {
    color: _materialDashboardReact.warningColor[0]
  },
  dangerText: {
    color: _materialDashboardReact.dangerColor[0]
  }
};
var _default = typographyStyle;
exports.default = _default;
