// material-dashboard-react/dist/components/Snackbar/Snackbar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _Snackbar = _interopRequireDefault(require("@material-ui/core/Snackbar"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));

var _snackbarContentStyle = _interopRequireDefault(require("assets/jss/material-dashboard-react/components/snackbarContentStyle.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function Snackbar(_ref) {
  var props = _extends({}, _ref);

  var classes = props.classes,
      message = props.message,
      color = props.color,
      close = props.close,
      icon = props.icon,
      place = props.place,
      open = props.open,
      rtlActive = props.rtlActive;
  var action = [];
  var messageClasses = (0, _classnames.default)(_defineProperty({}, classes.iconMessage, icon !== undefined));

  if (close !== undefined) {
    action = [_react.default.createElement(_IconButton.default, {
      className: classes.iconButton,
      key: "close",
      "aria-label": "Close",
      color: "inherit",
      onClick: function onClick() {
        return props.closeNotification();
      }
    }, _react.default.createElement(_Close.default, {
      className: classes.close
    }))];
  }

  return _react.default.createElement(_Snackbar.default, {
    anchorOrigin: {
      vertical: place.indexOf("t") === -1 ? "bottom" : "top",
      horizontal: place.indexOf("l") !== -1 ? "left" : place.indexOf("c") !== -1 ? "center" : "right"
    },
    open: open,
    message: _react.default.createElement("div", null, icon !== undefined ? _react.default.createElement(props.icon, {
      className: classes.icon
    }) : null, _react.default.createElement("span", {
      className: messageClasses
    }, message)),
    action: action,
    ContentProps: {
      classes: {
        root: classes.root + " " + classes[color],
        message: classes.message,
        action: (0, _classnames.default)(_defineProperty({}, classes.actionRTL, rtlActive))
      }
    }
  });
}

Snackbar.propTypes = {
  classes: _propTypes.default.object.isRequired,
  message: _propTypes.default.node.isRequired,
  color: _propTypes.default.oneOf(["info", "success", "warning", "danger", "primary"]),
  close: _propTypes.default.bool,
  icon: _propTypes.default.object,
  place: _propTypes.default.oneOf(["tl", "tr", "tc", "br", "bl", "bc"]),
  open: _propTypes.default.bool,
  rtlActive: _propTypes.default.bool,
  closeNotification: _propTypes.default.func
};

var _default = (0, _withStyles.default)(_snackbarContentStyle.default)(Snackbar);

exports.default = _default;
