// material-dashboard-react/dist/assets/jss/material-dashboard-react/layouts/rtlStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _materialDashboardReact = require("assets/jss/material-dashboard-react.jsx");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var appStyle = function appStyle(theme) {
  var _objectSpread2;

  return {
    wrapper: {
      position: "relative",
      top: "0",
      height: "100vh",
      direction: "rtl"
    },
    mainPanel: _objectSpread((_objectSpread2 = {}, _defineProperty(_objectSpread2, theme.breakpoints.up("md"), {
      width: "calc(100% - ".concat(_materialDashboardReact.drawerWidth, "px)")
    }), _defineProperty(_objectSpread2, "overflow", "auto"), _defineProperty(_objectSpread2, "position", "relative"), _defineProperty(_objectSpread2, "float", "left"), _objectSpread2), _materialDashboardReact.transition, {
      maxHeight: "100%",
      width: "100%",
      overflowScrolling: "touch"
    }),
    content: {
      marginTop: "70px",
      padding: "30px 15px",
      minHeight: "calc(100vh - 123px)"
    },
    container: _materialDashboardReact.container,
    map: {
      marginTop: "70px"
    }
  };
};

var _default = appStyle;
exports.default = _default;
