// material-dashboard-react/dist/routes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _Dashboard = _interopRequireDefault(require("@material-ui/icons/Dashboard"));

var _Person = _interopRequireDefault(require("@material-ui/icons/Person"));

var _LibraryBooks = _interopRequireDefault(require("@material-ui/icons/LibraryBooks"));

var _BubbleChart = _interopRequireDefault(require("@material-ui/icons/BubbleChart"));

var _LocationOn = _interopRequireDefault(require("@material-ui/icons/LocationOn"));

var _Notifications = _interopRequireDefault(require("@material-ui/icons/Notifications"));

var _Unarchive = _interopRequireDefault(require("@material-ui/icons/Unarchive"));

var _Language = _interopRequireDefault(require("@material-ui/icons/Language"));

var _Dashboard2 = _interopRequireDefault(require("views/Dashboard/Dashboard.jsx"));

var _UserProfile = _interopRequireDefault(require("views/UserProfile/UserProfile.jsx"));

var _TableList = _interopRequireDefault(require("views/TableList/TableList.jsx"));

var _Typography = _interopRequireDefault(require("views/Typography/Typography.jsx"));

var _Icons = _interopRequireDefault(require("views/Icons/Icons.jsx"));

var _Maps = _interopRequireDefault(require("views/Maps/Maps.jsx"));

var _Notifications2 = _interopRequireDefault(require("views/Notifications/Notifications.jsx"));

var _UpgradeToPro = _interopRequireDefault(require("views/UpgradeToPro/UpgradeToPro.jsx"));

var _RTLPage = _interopRequireDefault(require("views/RTLPage/RTLPage.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// @material-ui/icons
// core components/views for Admin layout
// core components/views for RTL layout
var dashboardRoutes = [{
  path: "/dashboard",
  name: "Dashboard",
  rtlName: "لوحة القيادة",
  icon: _Dashboard.default,
  component: _Dashboard2.default,
  layout: "/admin"
}, {
  path: "/user",
  name: "User Profile",
  rtlName: "ملف تعريفي للمستخدم",
  icon: _Person.default,
  component: _UserProfile.default,
  layout: "/admin"
}, {
  path: "/table",
  name: "Table List",
  rtlName: "قائمة الجدول",
  icon: "content_paste",
  component: _TableList.default,
  layout: "/admin"
}, {
  path: "/typography",
  name: "Typography",
  rtlName: "طباعة",
  icon: _LibraryBooks.default,
  component: _Typography.default,
  layout: "/admin"
}, {
  path: "/icons",
  name: "Icons",
  rtlName: "الرموز",
  icon: _BubbleChart.default,
  component: _Icons.default,
  layout: "/admin"
}, {
  path: "/maps",
  name: "Maps",
  rtlName: "خرائط",
  icon: _LocationOn.default,
  component: _Maps.default,
  layout: "/admin"
}, {
  path: "/notifications",
  name: "Notifications",
  rtlName: "إخطارات",
  icon: _Notifications.default,
  component: _Notifications2.default,
  layout: "/admin"
}, {
  path: "/rtl-page",
  name: "RTL Support",
  rtlName: "پشتیبانی از راست به چپ",
  icon: _Language.default,
  component: _RTLPage.default,
  layout: "/rtl"
}, {
  path: "/upgrade-to-pro",
  name: "Upgrade To PRO",
  rtlName: "التطور للاحترافية",
  icon: _Unarchive.default,
  component: _UpgradeToPro.default,
  layout: "/admin"
}];
var _default = dashboardRoutes;
exports.default = _default;
