// material-dashboard-react/dist/views/UserProfile/UserProfile.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _InputLabel = _interopRequireDefault(require("@material-ui/core/InputLabel"));

var _GridItem = _interopRequireDefault(require("components/Grid/GridItem.jsx"));

var _GridContainer = _interopRequireDefault(require("components/Grid/GridContainer.jsx"));

var _CustomInput = _interopRequireDefault(require("components/CustomInput/CustomInput.jsx"));

var _Button = _interopRequireDefault(require("components/CustomButtons/Button.jsx"));

var _Card = _interopRequireDefault(require("components/Card/Card.jsx"));

var _CardHeader = _interopRequireDefault(require("components/Card/CardHeader.jsx"));

var _CardAvatar = _interopRequireDefault(require("components/Card/CardAvatar.jsx"));

var _CardBody = _interopRequireDefault(require("components/Card/CardBody.jsx"));

var _CardFooter = _interopRequireDefault(require("components/Card/CardFooter.jsx"));

var _marc = _interopRequireDefault(require("assets/img/faces/marc.jpg"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// nodejs library to set properties for components
// @material-ui/core components
// core components
var styles = {
  cardCategoryWhite: {
    color: "rgba(255,255,255,.62)",
    margin: "0",
    fontSize: "14px",
    marginTop: "0",
    marginBottom: "0"
  },
  cardTitleWhite: {
    color: "#FFFFFF",
    marginTop: "0px",
    minHeight: "auto",
    fontWeight: "300",
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: "3px",
    textDecoration: "none"
  }
};

function UserProfile(props) {
  var classes = props.classes;
  return _react.default.createElement("div", null, _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 8
  }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
    color: "primary"
  }, _react.default.createElement("h4", {
    className: classes.cardTitleWhite
  }, "Edit Profile"), _react.default.createElement("p", {
    className: classes.cardCategoryWhite
  }, "Complete your profile")), _react.default.createElement(_CardBody.default, null, _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 5
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "Company (disabled)",
    id: "company-disabled",
    formControlProps: {
      fullWidth: true
    },
    inputProps: {
      disabled: true
    }
  })), _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 3
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "Username",
    id: "username",
    formControlProps: {
      fullWidth: true
    }
  })), _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 4
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "Email address",
    id: "email-address",
    formControlProps: {
      fullWidth: true
    }
  }))), _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 6
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "First Name",
    id: "first-name",
    formControlProps: {
      fullWidth: true
    }
  })), _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 6
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "Last Name",
    id: "last-name",
    formControlProps: {
      fullWidth: true
    }
  }))), _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 4
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "City",
    id: "city",
    formControlProps: {
      fullWidth: true
    }
  })), _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 4
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "Country",
    id: "country",
    formControlProps: {
      fullWidth: true
    }
  })), _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 4
  }, _react.default.createElement(_CustomInput.default, {
    labelText: "Postal Code",
    id: "postal-code",
    formControlProps: {
      fullWidth: true
    }
  }))), _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 12
  }, _react.default.createElement(_InputLabel.default, {
    style: {
      color: "#AAAAAA"
    }
  }, "About me"), _react.default.createElement(_CustomInput.default, {
    labelText: "Lamborghini Mercy, Your chick she so thirsty, I'm in that two seat Lambo.",
    id: "about-me",
    formControlProps: {
      fullWidth: true
    },
    inputProps: {
      multiline: true,
      rows: 5
    }
  })))), _react.default.createElement(_CardFooter.default, null, _react.default.createElement(_Button.default, {
    color: "primary"
  }, "Update Profile")))), _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 4
  }, _react.default.createElement(_Card.default, {
    profile: true
  }, _react.default.createElement(_CardAvatar.default, {
    profile: true
  }, _react.default.createElement("a", {
    href: "#pablo",
    onClick: function onClick(e) {
      return e.preventDefault();
    }
  }, _react.default.createElement("img", {
    src: _marc.default,
    alt: "..."
  }))), _react.default.createElement(_CardBody.default, {
    profile: true
  }, _react.default.createElement("h6", {
    className: classes.cardCategory
  }, "CEO / CO-FOUNDER"), _react.default.createElement("h4", {
    className: classes.cardTitle
  }, "Alec Thompson"), _react.default.createElement("p", {
    className: classes.description
  }, "Don", "'", "t be scared of the truth because we need to restart the human foundation in truth And I love you like Kanye loves Kanye I love Rick Owens\u2019 bed design but the back is..."), _react.default.createElement(_Button.default, {
    color: "primary",
    round: true
  }, "Follow"))))));
}

UserProfile.propTypes = {
  classes: _propTypes.default.object
};

var _default = (0, _withStyles.default)(styles)(UserProfile);

exports.default = _default;
