// material-dashboard-react/dist/layouts/Admin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _perfectScrollbar = _interopRequireDefault(require("perfect-scrollbar"));

require("perfect-scrollbar/css/perfect-scrollbar.css");

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _Navbar = _interopRequireDefault(require("components/Navbars/Navbar.jsx"));

var _Footer = _interopRequireDefault(require("components/Footer/Footer.jsx"));

var _Sidebar = _interopRequireDefault(require("components/Sidebar/Sidebar.jsx"));

var _FixedPlugin = _interopRequireDefault(require("components/FixedPlugin/FixedPlugin.jsx"));

var _routes = _interopRequireDefault(require("routes.js"));

var _dashboardStyle = _interopRequireDefault(require("assets/jss/material-dashboard-react/layouts/dashboardStyle.jsx"));

var _sidebar = _interopRequireDefault(require("assets/img/sidebar-2.jpg"));

var _reactlogo = _interopRequireDefault(require("assets/img/reactlogo.png"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var ps;

var switchRoutes = _react.default.createElement(_reactRouterDom.Switch, null, _routes.default.map(function (prop, key) {
  if (prop.layout === "/admin") {
    return _react.default.createElement(_reactRouterDom.Route, {
      path: prop.layout + prop.path,
      component: prop.component,
      key: key
    });
  }

  return null;
}), _react.default.createElement(_reactRouterDom.Redirect, {
  from: "/admin",
  to: "/admin/dashboard"
}));

var Dashboard =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Dashboard, _React$Component);

  function Dashboard() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, Dashboard);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Dashboard)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      image: _sidebar.default,
      color: "blue",
      hasImage: true,
      fixedClasses: "dropdown show",
      mobileOpen: false
    });

    _defineProperty(_assertThisInitialized(_this), "mainPanel", _react.default.createRef());

    _defineProperty(_assertThisInitialized(_this), "handleImageClick", function (image) {
      _this.setState({
        image: image
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleColorClick", function (color) {
      _this.setState({
        color: color
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleFixedClick", function () {
      if (_this.state.fixedClasses === "dropdown") {
        _this.setState({
          fixedClasses: "dropdown show"
        });
      } else {
        _this.setState({
          fixedClasses: "dropdown"
        });
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handleDrawerToggle", function () {
      _this.setState({
        mobileOpen: !_this.state.mobileOpen
      });
    });

    _defineProperty(_assertThisInitialized(_this), "resizeFunction", function () {
      if (window.innerWidth >= 960) {
        _this.setState({
          mobileOpen: false
        });
      }
    });

    return _this;
  }

  _createClass(Dashboard, [{
    key: "getRoute",
    value: function getRoute() {
      return window.location.pathname !== "/admin/maps";
    }
  }, {
    key: "componentDidMount",
    value: function componentDidMount() {
      if (navigator.platform.indexOf("Win") > -1) {
        ps = new _perfectScrollbar.default(this.mainPanel.current);
      }

      window.addEventListener("resize", this.resizeFunction);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(e) {
      if (e.history.location.pathname !== e.location.pathname) {
        this.mainPanel.current.scrollTop = 0;

        if (this.state.mobileOpen) {
          this.setState({
            mobileOpen: false
          });
        }
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      if (navigator.platform.indexOf("Win") > -1) {
        ps.destroy();
      }

      window.removeEventListener("resize", this.resizeFunction);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          classes = _this$props.classes,
          rest = _objectWithoutProperties(_this$props, ["classes"]);

      return _react.default.createElement("div", {
        className: classes.wrapper
      }, _react.default.createElement(_Sidebar.default, _extends({
        routes: _routes.default,
        logoText: "Creative Tim",
        logo: _reactlogo.default,
        image: this.state.image,
        handleDrawerToggle: this.handleDrawerToggle,
        open: this.state.mobileOpen,
        color: this.state.color
      }, rest)), _react.default.createElement("div", {
        className: classes.mainPanel,
        ref: this.mainPanel
      }, _react.default.createElement(_Navbar.default, _extends({
        routes: _routes.default,
        handleDrawerToggle: this.handleDrawerToggle
      }, rest)), this.getRoute() ? _react.default.createElement("div", {
        className: classes.content
      }, _react.default.createElement("div", {
        className: classes.container
      }, switchRoutes)) : _react.default.createElement("div", {
        className: classes.map
      }, switchRoutes), this.getRoute() ? _react.default.createElement(_Footer.default, null) : null, _react.default.createElement(_FixedPlugin.default, {
        handleImageClick: this.handleImageClick,
        handleColorClick: this.handleColorClick,
        bgColor: this.state["color"],
        bgImage: this.state["image"],
        handleFixedClick: this.handleFixedClick,
        fixedClasses: this.state.fixedClasses
      })));
    }
  }]);

  return Dashboard;
}(_react.default.Component);

Dashboard.propTypes = {
  classes: _propTypes.default.object.isRequired
};

var _default = (0, _withStyles.default)(_dashboardStyle.default)(Dashboard);

exports.default = _default;
