// material-dashboard-react/dist/views/UpgradeToPro/UpgradeToPro.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));

var _Check = _interopRequireDefault(require("@material-ui/icons/Check"));

var _GridItem = _interopRequireDefault(require("components/Grid/GridItem.jsx"));

var _GridContainer = _interopRequireDefault(require("components/Grid/GridContainer.jsx"));

var _Danger = _interopRequireDefault(require("components/Typography/Danger.jsx"));

var _Success = _interopRequireDefault(require("components/Typography/Success.jsx"));

var _Button = _interopRequireDefault(require("components/CustomButtons/Button.jsx"));

var _Card = _interopRequireDefault(require("components/Card/Card.jsx"));

var _CardHeader = _interopRequireDefault(require("components/Card/CardHeader.jsx"));

var _CardBody = _interopRequireDefault(require("components/Card/CardBody.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// nodejs library to set properties for components
// @material-ui/core components
// @material-ui/icons
// core components
var styles = {
  cardCategoryWhite: {
    "&,& a,& a:hover,& a:focus": {
      color: "rgba(255,255,255,.62)",
      margin: "0",
      fontSize: "14px",
      marginTop: "0",
      marginBottom: "0"
    },
    "& a,& a:hover,& a:focus": {
      color: "#FFFFFF"
    }
  },
  cardTitleWhite: {
    color: "#FFFFFF",
    marginTop: "0px",
    minHeight: "auto",
    fontWeight: "300",
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: "3px",
    textDecoration: "none",
    "& small": {
      color: "#777",
      fontSize: "65%",
      fontWeight: "400",
      lineHeight: "1"
    }
  },
  tableUpgradeWrapper: {
    display: "block",
    width: "100%",
    overflowX: "auto",
    WebkitOverflowScrolling: "touch",
    MsOverflowStyle: "-ms-autohiding-scrollbar"
  },
  table: {
    width: "100%",
    maxWidth: "100%",
    marginBottom: "1rem",
    backgroundColor: "transparent",
    borderCollapse: "collapse",
    display: "table",
    borderSpacing: "2px",
    borderColor: "grey",
    "& thdead tr th": {
      fontSize: "1.063rem",
      padding: "12px 8px",
      verticalAlign: "middle",
      fontWeight: "300",
      borderTopWidth: "0",
      borderBottom: "1px solid rgba(0, 0, 0, 0.06)",
      textAlign: "inherit"
    },
    "& tbody tr td": {
      padding: "12px 8px",
      verticalAlign: "middle",
      borderTop: "1px solid rgba(0, 0, 0, 0.06)"
    },
    "& td, & th": {
      display: "table-cell"
    }
  },
  center: {
    textAlign: "center"
  }
};

function UpgradeToPro(props) {
  var classes = props.classes;
  return _react.default.createElement(_GridContainer.default, {
    justify: "center"
  }, _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 8
  }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
    color: "info"
  }, _react.default.createElement("h4", {
    className: classes.cardTitleWhite
  }, "Material Dashboard PRO React"), _react.default.createElement("p", {
    className: classes.cardCategoryWhite
  }, "Are you looking for more components? Please check our Premium Version of Material Dashboard Angular.")), _react.default.createElement(_CardBody.default, null, _react.default.createElement("div", {
    className: classes.tableUpgradeWrapper
  }, _react.default.createElement("table", {
    className: classes.table
  }, _react.default.createElement("thead", null, _react.default.createElement("tr", null, _react.default.createElement("th", null), _react.default.createElement("th", {
    className: classes.center
  }, "Free"), _react.default.createElement("th", {
    className: classes.center
  }, "PRO"))), _react.default.createElement("tbody", null, _react.default.createElement("tr", null, _react.default.createElement("td", null, "Components"), _react.default.createElement("td", {
    className: classes.center
  }, "30"), _react.default.createElement("td", {
    className: classes.center
  }, "200")), _react.default.createElement("tr", null, _react.default.createElement("td", null, "Plugins"), _react.default.createElement("td", {
    className: classes.center
  }, "2"), _react.default.createElement("td", {
    className: classes.center
  }, "10")), _react.default.createElement("tr", null, _react.default.createElement("td", null, "Example Pages"), _react.default.createElement("td", {
    className: classes.center
  }, "7"), _react.default.createElement("td", {
    className: classes.center
  }, "28")), _react.default.createElement("tr", null, _react.default.createElement("td", null, "Login, Register, Pricing, Lock Pages"), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Danger.default, null, _react.default.createElement(_Close.default, null))), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Success.default, null, _react.default.createElement(_Check.default, null)))), _react.default.createElement("tr", null, _react.default.createElement("td", null, "ReactTables, ReactVectorMap, ReactSweetAlert, Wizard, Validation, ReactBigCalendar etc..."), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Danger.default, null, _react.default.createElement(_Close.default, null))), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Success.default, null, _react.default.createElement(_Check.default, null)))), _react.default.createElement("tr", null, _react.default.createElement("td", null, "Mini Sidebar"), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Danger.default, null, _react.default.createElement(_Close.default, null))), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Success.default, null, _react.default.createElement(_Check.default, null)))), _react.default.createElement("tr", null, _react.default.createElement("td", null, "Premium Support"), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Danger.default, null, _react.default.createElement(_Close.default, null))), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Success.default, null, _react.default.createElement(_Check.default, null)))), _react.default.createElement("tr", null, _react.default.createElement("td", null), _react.default.createElement("td", {
    className: classes.center
  }, "Free"), _react.default.createElement("td", {
    className: classes.center
  }, "Just $59")), _react.default.createElement("tr", null, _react.default.createElement("td", null), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Button.default, {
    round: true,
    disabled: true
  }, "Current Version")), _react.default.createElement("td", {
    className: classes.center
  }, _react.default.createElement(_Button.default, {
    round: true,
    color: "danger",
    href: "https://www.creative-tim.com/product/material-dashboard-pro-react?ref=mdr-upgrade-live"
  }, "Upgrade to Pro"))))))))));
}

UpgradeToPro.propTypes = {
  classes: _propTypes.default.object
};

var _default = (0, _withStyles.default)(styles)(UpgradeToPro);

exports.default = _default;
