// material-dashboard-react/dist/views/Typography/Typography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _Quote = _interopRequireDefault(require("components/Typography/Quote.jsx"));

var _Muted = _interopRequireDefault(require("components/Typography/Muted.jsx"));

var _Primary = _interopRequireDefault(require("components/Typography/Primary.jsx"));

var _Info = _interopRequireDefault(require("components/Typography/Info.jsx"));

var _Success = _interopRequireDefault(require("components/Typography/Success.jsx"));

var _Warning = _interopRequireDefault(require("components/Typography/Warning.jsx"));

var _Danger = _interopRequireDefault(require("components/Typography/Danger.jsx"));

var _Card = _interopRequireDefault(require("components/Card/Card.jsx"));

var _CardHeader = _interopRequireDefault(require("components/Card/CardHeader.jsx"));

var _CardBody = _interopRequireDefault(require("components/Card/CardBody.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// nodejs library to set properties for components
// @material-ui/core components
// core components
var style = {
  typo: {
    paddingLeft: "25%",
    marginBottom: "40px",
    position: "relative"
  },
  note: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    bottom: "10px",
    color: "#c0c1c2",
    display: "block",
    fontWeight: "400",
    fontSize: "13px",
    lineHeight: "13px",
    left: "0",
    marginLeft: "20px",
    position: "absolute",
    width: "260px"
  },
  cardCategoryWhite: {
    color: "rgba(255,255,255,.62)",
    margin: "0",
    fontSize: "14px",
    marginTop: "0",
    marginBottom: "0"
  },
  cardTitleWhite: {
    color: "#FFFFFF",
    marginTop: "0px",
    minHeight: "auto",
    fontWeight: "300",
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: "3px",
    textDecoration: "none"
  }
};

function TypographyPage(props) {
  var classes = props.classes;
  return _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
    color: "primary"
  }, _react.default.createElement("h4", {
    className: classes.cardTitleWhite
  }, "Material Dashboard Heading"), _react.default.createElement("p", {
    className: classes.cardCategoryWhite
  }, "Created using Roboto Font Family")), _react.default.createElement(_CardBody.default, null, _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Header 1"), _react.default.createElement("h1", null, "The Life of Material Dashboard")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Header 2"), _react.default.createElement("h2", null, "The Life of Material Dashboard")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Header 3"), _react.default.createElement("h3", null, "The Life of Material Dashboard")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Header 4"), _react.default.createElement("h4", null, "The Life of Material Dashboard")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Header 5"), _react.default.createElement("h5", null, "The Life of Material Dashboard")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Header 6"), _react.default.createElement("h6", null, "The Life of Material Dashboard")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Paragraph"), _react.default.createElement("p", null, "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers. I understand culture. I am the nucleus. I think that\u2019s a responsibility that I have, to push possibilities, to show people, this is the level that things could be at.")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Quote"), _react.default.createElement(_Quote.default, {
    text: "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers. I understand culture. I am the nucleus. I think that\u2019s a responsibility that I have, to push possibilities, to show people, this is the level that things could be at.",
    author: " Kanye West, Musician"
  })), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Muted Text"), _react.default.createElement(_Muted.default, null, "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers...")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Primary Text"), _react.default.createElement(_Primary.default, null, "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers...")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Info Text"), _react.default.createElement(_Info.default, null, "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers...")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Success Text"), _react.default.createElement(_Success.default, null, "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers...")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Warning Text"), _react.default.createElement(_Warning.default, null, "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers...")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Danger Text"), _react.default.createElement(_Danger.default, null, "I will be the leader of a company that ends up being worth billions of dollars, because I got the answers...")), _react.default.createElement("div", {
    className: classes.typo
  }, _react.default.createElement("div", {
    className: classes.note
  }, "Small Tag"), _react.default.createElement("h2", null, "Header with small subtitle", _react.default.createElement("br", null), _react.default.createElement("small", null, "Use ", '"', "Small", '"', " tag for the headers")))));
}

TypographyPage.propTypes = {
  classes: _propTypes.default.object
};

var _default = (0, _withStyles.default)(style)(TypographyPage);

exports.default = _default;
