// material-dashboard-react/dist/components/Navbars/AdminNavbarLinks.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _MenuItem = _interopRequireDefault(require("@material-ui/core/MenuItem"));

var _MenuList = _interopRequireDefault(require("@material-ui/core/MenuList"));

var _Grow = _interopRequireDefault(require("@material-ui/core/Grow"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _ClickAwayListener = _interopRequireDefault(require("@material-ui/core/ClickAwayListener"));

var _Hidden = _interopRequireDefault(require("@material-ui/core/Hidden"));

var _Popper = _interopRequireDefault(require("@material-ui/core/Popper"));

var _Divider = _interopRequireDefault(require("@material-ui/core/Divider"));

var _Person = _interopRequireDefault(require("@material-ui/icons/Person"));

var _Notifications = _interopRequireDefault(require("@material-ui/icons/Notifications"));

var _Dashboard = _interopRequireDefault(require("@material-ui/icons/Dashboard"));

var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));

var _CustomInput = _interopRequireDefault(require("components/CustomInput/CustomInput.jsx"));

var _Button = _interopRequireDefault(require("components/CustomButtons/Button.jsx"));

var _headerLinksStyle = _interopRequireDefault(require("assets/jss/material-dashboard-react/components/headerLinksStyle.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var AdminNavbarLinks =
/*#__PURE__*/
function (_React$Component) {
  _inherits(AdminNavbarLinks, _React$Component);

  function AdminNavbarLinks() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, AdminNavbarLinks);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(AdminNavbarLinks)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      openNotifcation: false,
      openProfile: false
    });

    _defineProperty(_assertThisInitialized(_this), "handleToggleNotification", function () {
      _this.setState(function (state) {
        return {
          openNotifcation: !state.openNotifcation
        };
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleCloseNotification", function (event) {
      if (_this.anchorNotification.contains(event.target)) {
        return;
      }

      _this.setState({
        openNotifcation: false
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleToggleProfile", function () {
      _this.setState(function (state) {
        return {
          openProfile: !state.openProfile
        };
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleCloseProfile", function (event) {
      if (_this.anchorProfile.contains(event.target)) {
        return;
      }

      _this.setState({
        openProfile: false
      });
    });

    return _this;
  }

  _createClass(AdminNavbarLinks, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var classes = this.props.classes;
      var _this$state = this.state,
          openNotifcation = _this$state.openNotifcation,
          openProfile = _this$state.openProfile;
      return _react.default.createElement("div", null, _react.default.createElement("div", {
        className: classes.searchWrapper
      }, _react.default.createElement(_CustomInput.default, {
        formControlProps: {
          className: classes.margin + " " + classes.search
        },
        inputProps: {
          placeholder: "Search",
          inputProps: {
            "aria-label": "Search"
          }
        }
      }), _react.default.createElement(_Button.default, {
        color: "white",
        "aria-label": "edit",
        justIcon: true,
        round: true
      }, _react.default.createElement(_Search.default, null))), _react.default.createElement(_Button.default, {
        color: window.innerWidth > 959 ? "transparent" : "white",
        justIcon: window.innerWidth > 959,
        simple: !(window.innerWidth > 959),
        "aria-label": "Dashboard",
        className: classes.buttonLink
      }, _react.default.createElement(_Dashboard.default, {
        className: classes.icons
      }), _react.default.createElement(_Hidden.default, {
        mdUp: true,
        implementation: "css"
      }, _react.default.createElement("p", {
        className: classes.linkText
      }, "Dashboard"))), _react.default.createElement("div", {
        className: classes.manager
      }, _react.default.createElement(_Button.default, {
        buttonRef: function buttonRef(node) {
          _this2.anchorNotification = node;
        },
        color: window.innerWidth > 959 ? "transparent" : "white",
        justIcon: window.innerWidth > 959,
        simple: !(window.innerWidth > 959),
        "aria-owns": openNotifcation ? "notification-menu-list-grow" : null,
        "aria-haspopup": "true",
        onClick: this.handleToggleNotification,
        className: classes.buttonLink
      }, _react.default.createElement(_Notifications.default, {
        className: classes.icons
      }), _react.default.createElement("span", {
        className: classes.notifications
      }, "5"), _react.default.createElement(_Hidden.default, {
        mdUp: true,
        implementation: "css"
      }, _react.default.createElement("p", {
        onClick: this.handleClick,
        className: classes.linkText
      }, "Notification"))), _react.default.createElement(_Popper.default, {
        open: openNotifcation,
        anchorEl: this.anchorNotification,
        transition: true,
        disablePortal: true,
        className: (0, _classnames.default)(_defineProperty({}, classes.popperClose, !openNotifcation)) + " " + classes.popperNav
      }, function (_ref) {
        var TransitionProps = _ref.TransitionProps,
            placement = _ref.placement;
        return _react.default.createElement(_Grow.default, _extends({}, TransitionProps, {
          id: "notification-menu-list-grow",
          style: {
            transformOrigin: placement === "bottom" ? "center top" : "center bottom"
          }
        }), _react.default.createElement(_Paper.default, null, _react.default.createElement(_ClickAwayListener.default, {
          onClickAway: _this2.handleCloseNotification
        }, _react.default.createElement(_MenuList.default, {
          role: "menu"
        }, _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseNotification,
          className: classes.dropdownItem
        }, "Mike John responded to your email"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseNotification,
          className: classes.dropdownItem
        }, "You have 5 new tasks"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseNotification,
          className: classes.dropdownItem
        }, "You", "'", "re now friend with Andrew"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseNotification,
          className: classes.dropdownItem
        }, "Another Notification"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseNotification,
          className: classes.dropdownItem
        }, "Another One")))));
      })), _react.default.createElement("div", {
        className: classes.manager
      }, _react.default.createElement(_Button.default, {
        buttonRef: function buttonRef(node) {
          _this2.anchorProfile = node;
        },
        color: window.innerWidth > 959 ? "transparent" : "white",
        justIcon: window.innerWidth > 959,
        simple: !(window.innerWidth > 959),
        "aria-owns": openNotifcation ? "profile-menu-list-grow" : null,
        "aria-haspopup": "true",
        onClick: this.handleToggleProfile,
        className: classes.buttonLink
      }, _react.default.createElement(_Person.default, {
        className: classes.icons
      }), _react.default.createElement(_Hidden.default, {
        mdUp: true,
        implementation: "css"
      }, _react.default.createElement("p", {
        className: classes.linkText
      }, "Profile"))), _react.default.createElement(_Popper.default, {
        open: openProfile,
        anchorEl: this.anchorProfile,
        transition: true,
        disablePortal: true,
        className: (0, _classnames.default)(_defineProperty({}, classes.popperClose, !openProfile)) + " " + classes.popperNav
      }, function (_ref2) {
        var TransitionProps = _ref2.TransitionProps,
            placement = _ref2.placement;
        return _react.default.createElement(_Grow.default, _extends({}, TransitionProps, {
          id: "profile-menu-list-grow",
          style: {
            transformOrigin: placement === "bottom" ? "center top" : "center bottom"
          }
        }), _react.default.createElement(_Paper.default, null, _react.default.createElement(_ClickAwayListener.default, {
          onClickAway: _this2.handleCloseProfile
        }, _react.default.createElement(_MenuList.default, {
          role: "menu"
        }, _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseProfile,
          className: classes.dropdownItem
        }, "Profile"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseProfile,
          className: classes.dropdownItem
        }, "Settings"), _react.default.createElement(_Divider.default, {
          light: true
        }), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleCloseProfile,
          className: classes.dropdownItem
        }, "Logout")))));
      })));
    }
  }]);

  return AdminNavbarLinks;
}(_react.default.Component);

AdminNavbarLinks.propTypes = {
  classes: _propTypes.default.object
};

var _default = (0, _withStyles.default)(_headerLinksStyle.default)(AdminNavbarLinks);

exports.default = _default;
