// material-dashboard-react/dist/components/Navbars/RTLNavbarLinks.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _MenuItem = _interopRequireDefault(require("@material-ui/core/MenuItem"));

var _MenuList = _interopRequireDefault(require("@material-ui/core/MenuList"));

var _Grow = _interopRequireDefault(require("@material-ui/core/Grow"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _ClickAwayListener = _interopRequireDefault(require("@material-ui/core/ClickAwayListener"));

var _Hidden = _interopRequireDefault(require("@material-ui/core/Hidden"));

var _Popper = _interopRequireDefault(require("@material-ui/core/Popper"));

var _Person = _interopRequireDefault(require("@material-ui/icons/Person"));

var _Notifications = _interopRequireDefault(require("@material-ui/icons/Notifications"));

var _Dashboard = _interopRequireDefault(require("@material-ui/icons/Dashboard"));

var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));

var _CustomInput = _interopRequireDefault(require("components/CustomInput/CustomInput.jsx"));

var _Button = _interopRequireDefault(require("components/CustomButtons/Button.jsx"));

var _rtlHeaderLinksStyle = _interopRequireDefault(require("assets/jss/material-dashboard-react/components/rtlHeaderLinksStyle.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var RTLNavbarLinks =
/*#__PURE__*/
function (_React$Component) {
  _inherits(RTLNavbarLinks, _React$Component);

  function RTLNavbarLinks() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, RTLNavbarLinks);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(RTLNavbarLinks)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      open: false
    });

    _defineProperty(_assertThisInitialized(_this), "handleToggle", function () {
      _this.setState(function (state) {
        return {
          open: !state.open
        };
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleClose", function (event) {
      if (_this.anchorEl.contains(event.target)) {
        return;
      }

      _this.setState({
        open: false
      });
    });

    return _this;
  }

  _createClass(RTLNavbarLinks, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var classes = this.props.classes;
      var open = this.state.open;
      return _react.default.createElement("div", null, _react.default.createElement("div", {
        className: classes.searchWrapper
      }, _react.default.createElement(_CustomInput.default, {
        formControlProps: {
          className: classes.margin + " " + classes.search
        },
        inputProps: {
          placeholder: "جستجو...",
          inputProps: {
            "aria-label": "Search"
          }
        }
      }), _react.default.createElement(_Button.default, {
        color: "white",
        "aria-label": "edit",
        justIcon: true,
        round: true
      }, _react.default.createElement(_Search.default, null))), _react.default.createElement(_Button.default, {
        color: window.innerWidth > 959 ? "transparent" : "white",
        justIcon: window.innerWidth > 959,
        simple: !(window.innerWidth > 959),
        "aria-label": "Dashboard",
        className: classes.buttonLink
      }, _react.default.createElement(_Dashboard.default, {
        className: classes.icons
      }), _react.default.createElement(_Hidden.default, {
        mdUp: true,
        implementation: "css"
      }, _react.default.createElement("p", {
        className: classes.linkText
      }, "\u0622\u0645\u0627\u0631\u0647\u0627"))), _react.default.createElement("div", {
        className: classes.manager
      }, _react.default.createElement(_Button.default, {
        buttonRef: function buttonRef(node) {
          _this2.anchorEl = node;
        },
        color: window.innerWidth > 959 ? "transparent" : "white",
        justIcon: window.innerWidth > 959,
        simple: !(window.innerWidth > 959),
        "aria-owns": open ? "menu-list-grow" : null,
        "aria-haspopup": "true",
        onClick: this.handleToggle,
        className: classes.buttonLink
      }, _react.default.createElement(_Notifications.default, {
        className: classes.icons
      }), _react.default.createElement("span", {
        className: classes.notifications
      }, "\u06F5"), _react.default.createElement(_Hidden.default, {
        mdUp: true,
        implementation: "css"
      }, _react.default.createElement("p", {
        onClick: this.handleClick,
        className: classes.linkText
      }, "\u0627\u0639\u0644\u0627\u0646\u200C\u0647\u0627"))), _react.default.createElement(_Popper.default, {
        open: open,
        anchorEl: this.anchorEl,
        transition: true,
        disablePortal: true,
        className: (0, _classnames.default)(_defineProperty({}, classes.popperClose, !open)) + " " + classes.popperNav
      }, function (_ref) {
        var TransitionProps = _ref.TransitionProps,
            placement = _ref.placement;
        return _react.default.createElement(_Grow.default, _extends({}, TransitionProps, {
          id: "menu-list-grow",
          style: {
            transformOrigin: placement === "bottom" ? "center top" : "center bottom"
          }
        }), _react.default.createElement(_Paper.default, null, _react.default.createElement(_ClickAwayListener.default, {
          onClickAway: _this2.handleClose
        }, _react.default.createElement(_MenuList.default, {
          role: "menu"
        }, _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleClose,
          className: classes.dropdownItem
        }, "\u0645\u062D\u0645\u062F\u0631\u0636\u0627 \u0628\u0647 \u0627\u06CC\u0645\u06CC\u0644 \u0634\u0645\u0627 \u067E\u0627\u0633\u062E \u062F\u0627\u062F"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleClose,
          className: classes.dropdownItem
        }, "\u0634\u0645\u0627 \u06F5 \u0648\u0638\u06CC\u0641\u0647 \u062C\u062F\u06CC\u062F \u062F\u0627\u0631\u06CC\u062F"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleClose,
          className: classes.dropdownItem
        }, "\u0627\u0632 \u062D\u0627\u0644\u0627 \u0634\u0645\u0627 \u0628\u0627 \u0639\u0644\u06CC\u0631\u0636\u0627 \u062F\u0648\u0633\u062A \u0647\u0633\u062A\u06CC\u062F"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleClose,
          className: classes.dropdownItem
        }, "\u0627\u0639\u0644\u0627\u0646 \u062F\u06CC\u06AF\u0631"), _react.default.createElement(_MenuItem.default, {
          onClick: _this2.handleClose,
          className: classes.dropdownItem
        }, "\u0627\u0639\u0644\u0627\u0646 \u062F\u06CC\u06AF\u0631")))));
      })), _react.default.createElement(_Button.default, {
        color: window.innerWidth > 959 ? "transparent" : "white",
        justIcon: window.innerWidth > 959,
        simple: !(window.innerWidth > 959),
        "aria-label": "Person",
        className: classes.buttonLink
      }, _react.default.createElement(_Person.default, {
        className: classes.icons
      }), _react.default.createElement(_Hidden.default, {
        mdUp: true,
        implementation: "css"
      }, _react.default.createElement("p", {
        className: classes.linkText
      }, "\u062D\u0633\u0627\u0628 \u06A9\u0627\u0631\u0628\u0631\u06CC"))));
    }
  }]);

  return RTLNavbarLinks;
}(_react.default.Component);

RTLNavbarLinks.propTypes = {
  classes: _propTypes.default.object
};

var _default = (0, _withStyles.default)(_rtlHeaderLinksStyle.default)(RTLNavbarLinks);

exports.default = _default;
