// material-dashboard-react/dist/components/Table/Table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _Table = _interopRequireDefault(require("@material-ui/core/Table"));

var _TableHead = _interopRequireDefault(require("@material-ui/core/TableHead"));

var _TableRow = _interopRequireDefault(require("@material-ui/core/TableRow"));

var _TableBody = _interopRequireDefault(require("@material-ui/core/TableBody"));

var _TableCell = _interopRequireDefault(require("@material-ui/core/TableCell"));

var _tableStyle = _interopRequireDefault(require("assets/jss/material-dashboard-react/components/tableStyle.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function CustomTable(_ref) {
  var props = _extends({}, _ref);

  var classes = props.classes,
      tableHead = props.tableHead,
      tableData = props.tableData,
      tableHeaderColor = props.tableHeaderColor;
  return _react.default.createElement("div", {
    className: classes.tableResponsive
  }, _react.default.createElement(_Table.default, {
    className: classes.table
  }, tableHead !== undefined ? _react.default.createElement(_TableHead.default, {
    className: classes[tableHeaderColor + "TableHeader"]
  }, _react.default.createElement(_TableRow.default, {
    className: classes.tableHeadRow
  }, tableHead.map(function (prop, key) {
    return _react.default.createElement(_TableCell.default, {
      className: classes.tableCell + " " + classes.tableHeadCell,
      key: key
    }, prop);
  }))) : null, _react.default.createElement(_TableBody.default, null, tableData.map(function (prop, key) {
    return _react.default.createElement(_TableRow.default, {
      key: key,
      className: classes.tableBodyRow
    }, prop.map(function (prop, key) {
      return _react.default.createElement(_TableCell.default, {
        className: classes.tableCell,
        key: key
      }, prop);
    }));
  }))));
}

CustomTable.defaultProps = {
  tableHeaderColor: "gray"
};
CustomTable.propTypes = {
  classes: _propTypes.default.object.isRequired,
  tableHeaderColor: _propTypes.default.oneOf(["warning", "primary", "danger", "success", "info", "rose", "gray"]),
  tableHead: _propTypes.default.arrayOf(_propTypes.default.string),
  tableData: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.string))
};

var _default = (0, _withStyles.default)(_tableStyle.default)(CustomTable);

exports.default = _default;
