// material-dashboard-react/dist/views/RTLPage/RTLPage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactChartist = _interopRequireDefault(require("react-chartist"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _Icon = _interopRequireDefault(require("@material-ui/core/Icon"));

var _Store = _interopRequireDefault(require("@material-ui/icons/Store"));

var _Warning = _interopRequireDefault(require("@material-ui/icons/Warning"));

var _DateRange = _interopRequireDefault(require("@material-ui/icons/DateRange"));

var _LocalOffer = _interopRequireDefault(require("@material-ui/icons/LocalOffer"));

var _Update = _interopRequireDefault(require("@material-ui/icons/Update"));

var _ArrowUpward = _interopRequireDefault(require("@material-ui/icons/ArrowUpward"));

var _AccessTime = _interopRequireDefault(require("@material-ui/icons/AccessTime"));

var _Accessibility = _interopRequireDefault(require("@material-ui/icons/Accessibility"));

var _BugReport = _interopRequireDefault(require("@material-ui/icons/BugReport"));

var _Code = _interopRequireDefault(require("@material-ui/icons/Code"));

var _Cloud = _interopRequireDefault(require("@material-ui/icons/Cloud"));

var _Button = _interopRequireDefault(require("components/CustomButtons/Button.jsx"));

var _GridItem = _interopRequireDefault(require("components/Grid/GridItem.jsx"));

var _GridContainer = _interopRequireDefault(require("components/Grid/GridContainer.jsx"));

var _Table = _interopRequireDefault(require("components/Table/Table.jsx"));

var _Tasks = _interopRequireDefault(require("components/Tasks/Tasks.jsx"));

var _CustomTabs = _interopRequireDefault(require("components/CustomTabs/CustomTabs.jsx"));

var _Danger = _interopRequireDefault(require("components/Typography/Danger.jsx"));

var _Card = _interopRequireDefault(require("components/Card/Card.jsx"));

var _CardHeader = _interopRequireDefault(require("components/Card/CardHeader.jsx"));

var _CardAvatar = _interopRequireDefault(require("components/Card/CardAvatar.jsx"));

var _CardIcon = _interopRequireDefault(require("components/Card/CardIcon.jsx"));

var _CardBody = _interopRequireDefault(require("components/Card/CardBody.jsx"));

var _CardFooter = _interopRequireDefault(require("components/Card/CardFooter.jsx"));

var _SnackbarContent = _interopRequireDefault(require("components/Snackbar/SnackbarContent.jsx"));

var _charts = require("variables/charts.jsx");

var _rtlStyle = _interopRequireDefault(require("assets/jss/material-dashboard-react/views/rtlStyle.jsx"));

var _marc = _interopRequireDefault(require("assets/img/faces/marc.jpg"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var bugs = ["طراح گرافیک از این متن به عنوان عنصری از ترکیب بندی برای پر کردن؟", "	نخست از متن‌های آزمایشی و بی‌معنی استفاده می‌کنند تا صرفا به مشتری یا صاحب کار خود نشان دهند؟", "همان حال کار آنها به نوعی وابسته به متن می‌باشد", "	آنها با استفاده از محتویات ساختگی، صفحه گرافیکی خود را صفحه‌آرایی می‌کنند"];
var website = ["بعد از اینکه متن در آن قرار گیرد چگونه به نظر می‌رسد و قلم‌ها و اندازه‌بندی‌ها چگونه در نظر گرفته", "اولیه شکل ظاهری و کلی طرح سفارش گرفته شده استفاده می نماید؟"];
var server = ["گرافیکی نشانگر چگونگی نوع و اندازه فونت و ظاهر متن باشد. معمولا طراحان گرافیک برای صفحه‌آرایی، نخست از متن‌های آزمایشی؟", "از این متن به عنوان عنصری از ترکیب بندی برای پر کردن صفحه و ارایه اولیه شکل ظاهری و کلی طرح سفارش گرفته شده استفاده می نماید، تا از نظر گرافیکی نشانگر چگونگی نوع و اندازه فونت و ظاهر متن باشد. معمولا طراحان گرافیک برای صفحه‌آرایی، نخست از متن‌های آزمایشی ؟", "از متن‌های آزمایشی و بی‌معنی استفاده می‌کنند تا صرفا به مشتری یا صاحب کار خود نشان دهند؟"];

var RTLPage =
/*#__PURE__*/
function (_React$Component) {
  _inherits(RTLPage, _React$Component);

  function RTLPage() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, RTLPage);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(RTLPage)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      value: 0
    });

    _defineProperty(_assertThisInitialized(_this), "handleChange", function (event, value) {
      _this.setState({
        value: value
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleChangeIndex", function (index) {
      _this.setState({
        value: index
      });
    });

    return _this;
  }

  _createClass(RTLPage, [{
    key: "render",
    value: function render() {
      var classes = this.props.classes;
      return _react.default.createElement("div", null, _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 6,
        md: 3
      }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
        color: "warning",
        stats: true,
        icon: true
      }, _react.default.createElement(_CardIcon.default, {
        color: "warning"
      }, _react.default.createElement(_Icon.default, null, "content_copy")), _react.default.createElement("p", {
        className: classes.cardCategory
      }, "\u0641\u0636\u0627 \u0645\u0635\u0631\u0641 \u0634\u062F\u0647"), _react.default.createElement("h3", {
        className: classes.cardTitle
      }, "49/50 ", _react.default.createElement("small", null, "GB"))), _react.default.createElement(_CardFooter.default, {
        stats: true
      }, _react.default.createElement("div", {
        className: classes.stats
      }, _react.default.createElement(_Danger.default, null, _react.default.createElement(_Warning.default, null)), _react.default.createElement("a", {
        href: "#pablo",
        onClick: function onClick(e) {
          return e.preventDefault();
        }
      }, "\u0641\u0636\u0627\u06CC \u0628\u06CC\u0634\u062A\u0631\u06CC \u062F\u0627\u0634\u062A\u0647 \u0628\u0627\u0634\u06CC\u062F..."))))), _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 6,
        md: 3
      }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
        color: "success",
        stats: true,
        icon: true
      }, _react.default.createElement(_CardIcon.default, {
        color: "success"
      }, _react.default.createElement(_Store.default, null)), _react.default.createElement("p", {
        className: classes.cardCategory
      }, "\u0633\u0648\u062F"), _react.default.createElement("h3", {
        className: classes.cardTitle
      }, "$34,245")), _react.default.createElement(_CardFooter.default, {
        stats: true
      }, _react.default.createElement("div", {
        className: classes.stats
      }, _react.default.createElement(_DateRange.default, null), "\u06F2\u06F4 \u0633\u0627\u0639\u062A \u0627\u062E\u06CC\u0631")))), _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 6,
        md: 3
      }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
        color: "danger",
        stats: true,
        icon: true
      }, _react.default.createElement(_CardIcon.default, {
        color: "danger"
      }, _react.default.createElement(_Icon.default, null, "info_outline")), _react.default.createElement("p", {
        className: classes.cardCategory
      }, "\u0645\u0634\u06A9\u0644\u0627\u062A \u062D\u0644 \u0634\u062F\u0647"), _react.default.createElement("h3", {
        className: classes.cardTitle
      }, "75")), _react.default.createElement(_CardFooter.default, {
        stats: true
      }, _react.default.createElement("div", {
        className: classes.stats
      }, _react.default.createElement(_LocalOffer.default, null), "\u062A\u0648\u0633\u0637 \u06AF\u06CC\u062A\u200C\u0647\u0627\u0628")))), _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 6,
        md: 3
      }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
        color: "info",
        stats: true,
        icon: true
      }, _react.default.createElement(_CardIcon.default, {
        color: "info"
      }, _react.default.createElement(_Accessibility.default, null)), _react.default.createElement("p", {
        className: classes.cardCategory
      }, "\u062F\u0646\u0628\u0627\u0644\u200C\u06A9\u0646\u0646\u062F\u0647"), _react.default.createElement("h3", {
        className: classes.cardTitle
      }, "+245")), _react.default.createElement(_CardFooter.default, {
        stats: true
      }, _react.default.createElement("div", {
        className: classes.stats
      }, _react.default.createElement(_Update.default, null), "\u0647\u0645\u200C\u0627\u06A9\u0646\u0648\u0646"))))), _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 12,
        md: 4
      }, _react.default.createElement(_Card.default, {
        chart: true
      }, _react.default.createElement(_CardHeader.default, {
        color: "success"
      }, _react.default.createElement(_reactChartist.default, {
        className: "ct-chart",
        data: _charts.dailySalesChart.data,
        type: "Line",
        options: _charts.dailySalesChart.options,
        listener: _charts.dailySalesChart.animation
      })), _react.default.createElement(_CardBody.default, null, _react.default.createElement("h4", {
        className: classes.cardTitle
      }, "\u0641\u0631\u0648\u0634 \u0631\u0648\u0632\u0627\u0646\u0647"), _react.default.createElement("p", {
        className: classes.cardCategory
      }, _react.default.createElement("span", {
        className: classes.successText
      }, _react.default.createElement(_ArrowUpward.default, {
        className: classes.upArrowCardCategory
      }), " 55%"), " ", "\u0631\u0634\u062F \u062F\u0631 \u0641\u0631\u0648\u0634 \u0627\u0645\u0631\u0648\u0632.")), _react.default.createElement(_CardFooter.default, {
        chart: true
      }, _react.default.createElement("div", {
        className: classes.stats
      }, _react.default.createElement(_AccessTime.default, null), " \u06F4 \u062F\u0642\u06CC\u0642\u0647 \u067E\u06CC\u0634")))), _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 12,
        md: 4
      }, _react.default.createElement(_Card.default, {
        chart: true
      }, _react.default.createElement(_CardHeader.default, {
        color: "warning"
      }, _react.default.createElement(_reactChartist.default, {
        className: "ct-chart",
        data: _charts.emailsSubscriptionChart.data,
        type: "Bar",
        options: _charts.emailsSubscriptionChart.options,
        responsiveOptions: _charts.emailsSubscriptionChart.responsiveOptions,
        listener: _charts.emailsSubscriptionChart.animation
      })), _react.default.createElement(_CardBody.default, null, _react.default.createElement("h4", {
        className: classes.cardTitle
      }, "\u062F\u0646\u0628\u0627\u0644 \u06A9\u0646\u0646\u062F\u0647\u200C\u0647\u0627\u06CC \u0627\u06CC\u0645\u06CC\u0644\u06CC"), _react.default.createElement("p", {
        className: classes.cardCategory
      }, "\u06A9\u0627\u0631\u0627\u06CC\u06CC \u0622\u062E\u0631\u06CC\u0646 \u06A9\u0645\u067E\u06CC\u0646")), _react.default.createElement(_CardFooter.default, {
        chart: true
      }, _react.default.createElement("div", {
        className: classes.stats
      }, _react.default.createElement(_AccessTime.default, null), " \u06A9\u0645\u067E\u06CC\u0646 \u062F\u0648 \u0631\u0648\u0632 \u067E\u06CC\u0634 \u0627\u0631\u0633\u0627\u0644 \u0634\u062F")))), _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 12,
        md: 4
      }, _react.default.createElement(_Card.default, {
        chart: true
      }, _react.default.createElement(_CardHeader.default, {
        color: "danger"
      }, _react.default.createElement(_reactChartist.default, {
        className: "ct-chart",
        data: _charts.completedTasksChart.data,
        type: "Line",
        options: _charts.completedTasksChart.options,
        listener: _charts.completedTasksChart.animation
      })), _react.default.createElement(_CardBody.default, null, _react.default.createElement("h4", {
        className: classes.cardTitle
      }, "\u0648\u0638\u0627\u06CC\u0641 \u0627\u0646\u062C\u0627\u0645 \u0634\u062F\u0647"), _react.default.createElement("p", {
        className: classes.cardCategory
      }, "\u06A9\u0627\u0631\u0627\u06CC\u06CC \u0622\u062E\u0631\u06CC\u0646 \u06A9\u0645\u067E\u06CC\u0646")), _react.default.createElement(_CardFooter.default, {
        chart: true
      }, _react.default.createElement("div", {
        className: classes.stats
      }, _react.default.createElement(_AccessTime.default, null), " \u06A9\u0645\u067E\u06CC\u0646 \u062F\u0648 \u0631\u0648\u0632 \u067E\u06CC\u0634 \u0627\u0631\u0633\u0627\u0644 \u0634\u062F"))))), _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 12,
        md: 6
      }, _react.default.createElement(_CustomTabs.default, {
        title: "\u0648\u0638\u0627\u06CC\u0641:",
        headerColor: "primary",
        rtlActive: true,
        tabs: [{
          tabName: "باگ‌ها",
          tabIcon: _BugReport.default,
          tabContent: _react.default.createElement(_Tasks.default, {
            checkedIndexes: [0, 3],
            tasksIndexes: [0, 1, 2, 3],
            tasks: bugs,
            rtlActive: true
          })
        }, {
          tabName: "وبسایت",
          tabIcon: _Code.default,
          tabContent: _react.default.createElement(_Tasks.default, {
            checkedIndexes: [0],
            tasksIndexes: [0, 1],
            tasks: website,
            rtlActive: true
          })
        }, {
          tabName: "سرور",
          tabIcon: _Cloud.default,
          tabContent: _react.default.createElement(_Tasks.default, {
            checkedIndexes: [1],
            tasksIndexes: [0, 1, 2],
            tasks: server,
            rtlActive: true
          })
        }]
      })), _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 12,
        md: 6
      }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
        color: "warning"
      }, _react.default.createElement("h4", {
        className: classes.cardTitleWhite
      }, "\u0622\u0645\u0627\u0631 \u06A9\u0627\u0631\u06A9\u0646\u0627\u0646"), _react.default.createElement("p", {
        className: classes.cardCategoryWhite
      }, "\u06A9\u0627\u0631\u06A9\u0646\u0627\u0646 \u062C\u062F\u06CC\u062F \u0627\u0632 \u06F1\u06F5 \u0622\u0628\u0627\u0646 \u06F1\u06F3\u06F9\u06F6")), _react.default.createElement(_CardBody.default, null, _react.default.createElement(_Table.default, {
        tableHeaderColor: "warning",
        tableHead: ["کد", "نام", "حقوق", "استان"],
        tableData: [["1", "احمد حسینی	", "$36,738", "مازندران"], ["2", "مینا رضایی	", "$23,789", "گلستان"], ["3", "مبینا احمدپور	", "$56,142", "تهران"], ["4", "جلال آقایی	", "$38,735", "شهرکرد"]]
      }))))), _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 12,
        md: 6
      }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
        color: "primary"
      }, _react.default.createElement("h4", {
        className: classes.cardTitleWhite
      }, "\u0627\u0639\u0644\u0627\u0646 \u0647\u0627"), _react.default.createElement("p", {
        className: classes.cardCategoryWhite
      }, "\u064A\u062F\u0648\u064A\u0627 \u0645\u0646 \u0642\u0628\u0644 \u0623\u0635\u062F\u0642\u0627\u0626\u0646\u0627 \u0645\u0646", " ", _react.default.createElement("a", {
        target: "_blank",
        href: "https://material-ui-next.com/?ref=creativetime"
      }, "\u0648\u0627\u062C\u0647\u0629 \u0627\u0644\u0645\u0633\u062A\u062E\u062F\u0645 \u0627\u0644\u0645\u0627\u062F\u064A\u0629"), " ", "\u0648\u0646\u0635\u0628 \u0645\u0646 \u0642\u0628\u0644", " ", _react.default.createElement("a", {
        target: "_blank",
        href: "https://www.creative-tim.com/?ref=mdr-rtl-page"
      }, "\u0627\u0644\u0625\u0628\u062F\u0627\u0639\u064A\u0629 \u062A\u064A\u0645"), ". \u064A\u0631\u062C\u0649 \u0627\u0644\u062A\u062D\u0642\u0642 \u0645\u0646", " ", _react.default.createElement("a", {
        href: "#pablo",
        target: "_blank"
      }, "\u0648\u062B\u0627\u0626\u0642 \u0643\u0627\u0645\u0644\u0629"), ".")), _react.default.createElement(_CardBody.default, null, _react.default.createElement(_SnackbarContent.default, {
        message: 'این یک اعلان است که با کلاس color="warning" ایجاد شده است.',
        close: true,
        rtlActive: true,
        color: "warning"
      }), _react.default.createElement(_SnackbarContent.default, {
        message: 'این یک اعلان است که با کلاس color="primary" ایجاد شده است.',
        close: true,
        rtlActive: true,
        color: "primary"
      }), _react.default.createElement(_SnackbarContent.default, {
        message: "این یک اعلان با دکمه بستن و آیکن است",
        close: true,
        rtlActive: true,
        color: "info"
      })))), _react.default.createElement(_GridItem.default, {
        xs: 12,
        sm: 12,
        md: 6
      }, _react.default.createElement(_Card.default, {
        profile: true
      }, _react.default.createElement(_CardAvatar.default, {
        profile: true
      }, _react.default.createElement("a", {
        href: "#pablo",
        onClick: function onClick(e) {
          return e.preventDefault();
        }
      }, _react.default.createElement("img", {
        src: _marc.default,
        alt: "..."
      }))), _react.default.createElement(_CardBody.default, {
        profile: true
      }, _react.default.createElement("h6", {
        className: classes.cardCategory
      }, "\u0645\u062F\u06CC\u0631\u0639\u0627\u0645\u0644 / \u0645\u062F\u06CC\u0631\u0641\u0646\u06CC"), _react.default.createElement("h4", {
        className: classes.cardTitle
      }, "\u062E\u062F\u0627\u062F\u0627\u062F \u0639\u0632\u06CC\u0632\u06CC"), _react.default.createElement("p", {
        className: classes.description
      }, "\u0637\u0631\u0627\u062D \u06AF\u0631\u0627\u0641\u06CC\u06A9 \u0627\u0632 \u0627\u06CC\u0646 \u0645\u062A\u0646 \u0628\u0647 \u0639\u0646\u0648\u0627\u0646 \u0639\u0646\u0635\u0631\u06CC \u0627\u0632 \u062A\u0631\u06A9\u06CC\u0628 \u0628\u0646\u062F\u06CC \u0628\u0631\u0627\u06CC \u067E\u0631 \u06A9\u0631\u062F\u0646 \u0635\u0641\u062D\u0647 \u0648 \u0627\u0631\u0627\u06CC\u0647 \u0627\u0648\u0644\u06CC\u0647 \u0634\u06A9\u0644 \u0638\u0627\u0647\u0631\u06CC \u0648 \u06A9\u0644\u06CC \u0637\u0631\u062D \u0633\u0641\u0627\u0631\u0634 \u06AF\u0631\u0641\u062A\u0647 \u0634\u062F\u0647 \u0627\u0633\u062A\u0641\u0627\u062F\u0647 \u0645\u06CC \u0646\u0645\u0627\u06CC\u062F\u060C \u062A\u0627 \u0627\u0632 \u0646\u0638\u0631 \u06AF\u0631\u0627\u0641\u06CC\u06A9\u06CC \u0646\u0634\u0627\u0646\u06AF\u0631 \u0686\u06AF\u0648\u0646\u06AF\u06CC \u0646\u0648\u0639 \u0648 \u0627\u0646\u062F\u0627\u0632\u0647 \u0641\u0648\u0646\u062A \u0648 \u0638\u0627\u0647\u0631 \u0645\u062A\u0646 \u0628\u0627\u0634\u062F. \u0645\u0639\u0645\u0648\u0644\u0627 \u0637\u0631\u0627\u062D\u0627\u0646 \u06AF\u0631\u0627\u0641\u06CC\u06A9 \u0628\u0631\u0627\u06CC \u0635\u0641\u062D\u0647\u200C\u0622\u0631\u0627\u06CC\u06CC\u060C \u0646\u062E\u0633\u062A \u0627\u0632 \u0645\u062A\u0646\u200C\u0647\u0627\u06CC \u0622\u0632\u0645\u0627\u06CC\u0634\u06CC \u0648 \u0628\u06CC\u200C\u0645\u0639\u0646\u06CC \u0627\u0633\u062A\u0641\u0627\u062F\u0647 \u0645\u06CC\u200C\u06A9\u0646\u0646\u062F ..."), _react.default.createElement(_Button.default, {
        color: "primary",
        round: true
      }, "\u062F\u0646\u0628\u0627\u0644\u200C\u06A9\u0631\u062F\u0646"))))));
    }
  }]);

  return RTLPage;
}(_react.default.Component);

RTLPage.propTypes = {
  classes: _propTypes.default.object.isRequired
};

var _default = (0, _withStyles.default)(_rtlStyle.default)(RTLPage);

exports.default = _default;
