// material-dashboard-react/dist/views/TableList/TableList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _withStyles = _interopRequireDefault(require("@material-ui/core/styles/withStyles"));

var _GridItem = _interopRequireDefault(require("components/Grid/GridItem.jsx"));

var _GridContainer = _interopRequireDefault(require("components/Grid/GridContainer.jsx"));

var _Table = _interopRequireDefault(require("components/Table/Table.jsx"));

var _Card = _interopRequireDefault(require("components/Card/Card.jsx"));

var _CardHeader = _interopRequireDefault(require("components/Card/CardHeader.jsx"));

var _CardBody = _interopRequireDefault(require("components/Card/CardBody.jsx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// nodejs library to set properties for components
// @material-ui/core components
// core components
var styles = {
  cardCategoryWhite: {
    "&,& a,& a:hover,& a:focus": {
      color: "rgba(255,255,255,.62)",
      margin: "0",
      fontSize: "14px",
      marginTop: "0",
      marginBottom: "0"
    },
    "& a,& a:hover,& a:focus": {
      color: "#FFFFFF"
    }
  },
  cardTitleWhite: {
    color: "#FFFFFF",
    marginTop: "0px",
    minHeight: "auto",
    fontWeight: "300",
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: "3px",
    textDecoration: "none",
    "& small": {
      color: "#777",
      fontSize: "65%",
      fontWeight: "400",
      lineHeight: "1"
    }
  }
};

function TableList(props) {
  var classes = props.classes;
  return _react.default.createElement(_GridContainer.default, null, _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 12
  }, _react.default.createElement(_Card.default, null, _react.default.createElement(_CardHeader.default, {
    color: "primary"
  }, _react.default.createElement("h4", {
    className: classes.cardTitleWhite
  }, "Simple Table"), _react.default.createElement("p", {
    className: classes.cardCategoryWhite
  }, "Here is a subtitle for this table")), _react.default.createElement(_CardBody.default, null, _react.default.createElement(_Table.default, {
    tableHeaderColor: "primary",
    tableHead: ["Name", "Country", "City", "Salary"],
    tableData: [["Dakota Rice", "Niger", "Oud-Turnhout", "$36,738"], ["Minerva Hooper", "Curaçao", "Sinaai-Waas", "$23,789"], ["Sage Rodriguez", "Netherlands", "Baileux", "$56,142"], ["Philip Chaney", "Korea, South", "Overland Park", "$38,735"], ["Doris Greene", "Malawi", "Feldkirchen in Kärnten", "$63,542"], ["Mason Porter", "Chile", "Gloucester", "$78,615"]]
  })))), _react.default.createElement(_GridItem.default, {
    xs: 12,
    sm: 12,
    md: 12
  }, _react.default.createElement(_Card.default, {
    plain: true
  }, _react.default.createElement(_CardHeader.default, {
    plain: true,
    color: "primary"
  }, _react.default.createElement("h4", {
    className: classes.cardTitleWhite
  }, "Table on Plain Background"), _react.default.createElement("p", {
    className: classes.cardCategoryWhite
  }, "Here is a subtitle for this table")), _react.default.createElement(_CardBody.default, null, _react.default.createElement(_Table.default, {
    tableHeaderColor: "primary",
    tableHead: ["ID", "Name", "Country", "City", "Salary"],
    tableData: [["1", "Dakota Rice", "$36,738", "Niger", "Oud-Turnhout"], ["2", "Minerva Hooper", "$23,789", "Curaçao", "Sinaai-Waas"], ["3", "Sage Rodriguez", "$56,142", "Netherlands", "Baileux"], ["4", "Philip Chaney", "$38,735", "Korea, South", "Overland Park"], ["5", "Doris Greene", "$63,542", "Malawi", "Feldkirchen in Kärnten"], ["6", "Mason Porter", "$78,615", "Chile", "Gloucester"]]
  })))));
}

TableList.propTypes = {
  classes: _propTypes.default.object
};

var _default = (0, _withStyles.default)(styles)(TableList);

exports.default = _default;
